src/extensions/agent_modules/cloud_communication/communication_channel/mqtt_broker/src/adu_communication_channel.c

```diff
@@ -664,35 +693,45 @@ void ADUC_Communication_Channel_OnPublish(
  */
 void ADUC_Communication_Channel_OnSubscribe(
     struct mosquitto* mosq,
-    void* commModuleHandle,
+    void* commChannelModuleHandle,
     int mid,
     int qos_count,
     const int* granted_qos,
     const mosquitto_property* props)
 {
     Log_Info("on_subscribe: Subscribed with mid %d; %d topics.", mid, qos_count);
-    ADUC_AGENT_MODULE_INTERFACE* interface = (ADUC_AGENT_MODULE_INTERFACE*)commModuleHandle;
-
-    if (interface == NULL)
-    {
-        Log_Error("Null parameter (interface=%p)", interface);
-        return;
-    }
-
-    ADU_MQTT_COMMUNICATION_MGR_STATE* commMgrState = (ADU_MQTT_COMMUNICATION_MGR_STATE*)interface->moduleData;
+    ADU_MQTT_COMMUNICATION_MGR_STATE* commMgrState =
+        CommunicationManagerStateFromModuleHandle(commChannelModuleHandle);
     if (commMgrState == NULL)
     {
-        Log_Error("commMgrState is NULL");
+        Log_Error("State is NULL. (handle=%p)", commChannelModuleHandle);
```

Comment:
Is this a real case? Seems like Log_Debug is more appropriate. Let us not log pointer addresses but use debugger instead